src/test/java/org/prebid/server/bidder/bematterfull/BematterfullBidderTest.java

```diff
@@ -0,0 +1,300 @@
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(1).first()
+                .satisfies(request -> assertThat(request.getUri())
+                        .isEqualTo("https://test-url.com?host=env&pid=pid"));
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnExpectedHeaders() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(1).first()
+                .extracting(HttpRequest::getHeaders)
+                .satisfies(headers -> assertThat(headers.get(CONTENT_TYPE_HEADER))
+                        .isEqualTo(APPLICATION_JSON_CONTENT_TYPE))
+                .satisfies(headers -> assertThat(headers.get(ACCEPT_HEADER))
+                        .isEqualTo(APPLICATION_JSON_VALUE));
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnExpectedBody() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> results = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(results.getValue()).hasSize(1).first()
+                .satisfies(request -> assertThat(request.getBody())
+                        .isEqualTo(jacksonMapper.encodeToBytes(bidRequest)))
+                .satisfies(request -> assertThat(request.getPayload())
+                        .isEqualTo(bidRequest));
+        assertThat(results.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnExpectedRequestMethod() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> results = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(results.getValue()).hasSize(1).first()
+                .satisfies(request -> assertThat(request.getMethod()).isEqualTo(HttpMethod.POST));
+        assertThat(results.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnExpectedImpIds() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> results = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(results.getValue()).hasSize(1).first()
+                .satisfies(request -> assertThat(request.getImpIds()).isEqualTo(Set.of("imp_id")));
+        assertThat(results.getErrors()).isEmpty();
+    }
```

Comment:
There is no need for such tests.

Reply:
For one of my PRs @SerhiiNahornyi required them

Reply:
Good test, as in future we will require this for each adapter